### DISCUSSION TITLE
HWACC CDM (Change Detection Module)

### DISCUSSION THREAD

**Discussion by CEGC2024** [HimaxWiseEyePlus/Seeed_Grove_Vision_AI_Module_V2#60] [General]
Hi all,

I've been experimenting with the WE2 and would like to make use of the CDM for motion detection. Is the CDM functional with the SDK at this point in time? I see it is included and presumably initialized with some of the DP path functions such as sensordplib_set_INT1_HWACC however hx_drv_cdm_get_enable always returns CDM_ENABLE_OFF.

I am also a bit confused about the memory/address allocation to WDMA1. Is this shared by both the HW2x2 and CDM? How do I access the meta data of the CDM?

Thank you.

**Comment by kris-himax** [comment]
Hi @CEGC2024 
1. If you want to use CDM, then SENSORDPLIB_PATH_E in cisdp_dp_init should be set so that the dp path passes through CDM.
`SENSORDPLIB_PATH_INP_HW2x2_CDM` or `SENSORDPLIB_PATH_INP_CDM` ... 
https://github.com/HimaxWiseEyePlus/Seeed_Grove_Vision_AI_Module_V2/blob/main/EPII_CM55M_APP_S/library/sensordp/inc/sensor_dp_lib.h#L184
2. You can reference following `SENSORDPLIB_PATH_E` about CDM. At WE2 hardware, the data path is HW2x2 -> CDM -> WDMA1.
If you enable CDM and you want to access the meta data of the CDM, you can just access WDMA1 and the HW2x2 result won't be stored at WDMA1 after CDM.
```
SENSORDPLIB_PATH_INP_HW2x2_CDM,   			/*!< Enum Datapath U2: Sensor --> INP --> 2x2 block --> CDM --> WDMA1. */
SENSORDPLIB_PATH_INP_CDM,					/*!< Enum Datapath D2: Sensor --> INP --> CDM --> WDMA1. */
```
